functions.sql

```diff
@@ -0,0 +1,68 @@
+/* Additional database functions for openstreetmap-carto */
+
+/* Access functions below adapted from https://github.com/imagico/osm-carto-alternative-colors/tree/591c861112b4e5d44badd108f4cd1409146bca0b/sql/roads.sql */
+
+/* Simplified 'yes', 'destination', 'no', 'unknown', NULL scale for access restriction 
+  'no' is returned if the rendering for highway category does not support 'restricted'.
+  NULL is functionally equivalent to 'yes', but indicates the absence of a restriction 
+  rather than a positive access = yes. 'unknown' corresponds to an uninterpretable 
+  access restriction e.g. access=unknown or motorcar=occasionally */
+CREATE OR REPLACE FUNCTION carto_int_access(accesstag text, allow_restricted boolean)
+	RETURNS text
+	LANGUAGE SQL
+	IMMUTABLE PARALLEL SAFE
+AS $$
+SELECT
+	CASE
+		WHEN accesstag IN ('yes', 'designated', 'permissive') THEN 'yes'
+		WHEN accesstag IN ('destination',  'delivery', 'customers') THEN
+			CASE WHEN allow_restricted = TRUE  THEN 'restricted' ELSE 'yes' END
+		WHEN accesstag IN ('no', 'permit', 'private', 'agricultural', 'forestry', 'agricultural;forestry') THEN 'no'
+		WHEN accesstag IS NULL THEN NULL
+		ELSE 'unknown'
```

Comment:
With how this is being called (`carto_int_access(NULLIF(mode, 'unknown'), "access", FALSE`) we have `'unknown'` being used in two different ways.

This can all be simplified because the return values of `'unknown'` and `NULL` are treated identically everywhere that I can see.

Reply:
Yes, I agree that `unknown` is being used in two different ways. It might be better to call this `unrecognised`.

> This can all be simplified because the return values of `'unknown'` and `NULL` are treated identically everywhere that I can see.

It is helpful to indicate that access tagging has not be recognised, independent of whether this is rendered or not. It is a distinct outcome which has been evaluated.